### ISSUE TITLE
Better error running basin_stats with a data_id not in catalog metadata

### ISSUE THREAD

**Issue by jsta** [VeinsOfTheEarth/rabpro#148]
Consider this scenario:

* a data source like GlobalSurfaceWater Dataset comes out with a new version
* the catalog watcher removes the old version from the catalog (it still exists on GEE)
* if you run a query against the old version, you will get a cryptic error (preceded way up the stack by a warning that the id is not in the catalog)

This is because the call to `Dataset` emits a warning (not an error) and passes on an empty output to `basin_stats.compute` because the id is not in the catalog. `compute` attempts to proceed as normal with the "empty" request. We should probably catch these in `compute` and throw an error.

**Comment by jonschwenk** [comment]
Should be easy enough to implement.

For this particular case (and maybe others like it), I think the solution would be to put deprecated assets in rabpro's user_added_catalog. This would avoid errors but this is probably a case where you want it to fail (if you weren't aware of the new version) so maybe that's a bad idea.

**Comment by jsta** [comment]
Should we enable users to override any checks against the catalog and just run the query? For reasons that are unclear to me, DAYMET does not appear in the scraped json but is clearly available in the GEE catalog.

**Comment by jonschwenk** [comment]
Yes, I think so. I was struggling with this for VotE forcings/characteristics fetching as well.

**Comment by jsta** [comment]
The two dataset attributes that you get "for free" if you entry is in the catalog is `type` and `resolution`. I've made it so you can skip dataset validation (with `validate_dataset_list=True`) and manually specify these values.
